src/resolve/resolved-link-bus.c

```diff
@@ -465,6 +465,10 @@ int bus_link_method_set_default_route(sd_bus_message *message, void *userdata, s
         if (l->default_route != b) {
                 l->default_route = b;
 
+                DnsServer *server = l->manager->current_dns_server;
+                if (server && server->type == DNS_SERVER_FALLBACK)
+                        manager_set_dns_server(l->manager, NULL);
```

Comment:
i'd prefer to avoid the "reachover" from the Link code here to the Manager code. maybe add manager_reconsider_current_server() or so which we just call here?

also comments would be good explaining what's going on here, because not obvious

Reply:
`manager_set_dns_server(m, NULL)` basically already is `manager_reconsider_current_server`. If you don't like the raw member access into manager, I think stuffing this logic in a `link_set_default_route` as suggested by @yuwata will handle that.